### PR TITLE
build: update Angular version to v17 prerelease

### PR DESCRIPTION
**Pull request by clydin**
Package dependency versions and the version used to generate a new project have been updated to use `17.0.0-next.1`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,22 +59,22 @@
   },
   "devDependencies": {
     "@ampproject/remapping": "2.2.1",
-    "@angular/animations": "16.1.8",
+    "@angular/animations": "17.0.0-next.0",
```
Comment (crisbeto) [inline package.json:62]: Drive-by comment: can we bump this to `next.1` which has support for TS 5.2?

**Review by alan-agius4 — APPROVED** [review]
Will merge to unblock CI, but can you do a followup to also update https://github.com/angular/angular-cli/blob/981f80e5a9946f4a7c967abe3d3b00821543eab1/packages/angular/ssr/package.json#L16? 

Thanks.
